### PR TITLE
[SYNPY-1207] Update property name

### PR DESCRIPTION
**Pull request by vpchung**
This will make it so all properties are consistent with styling; currently, `dataset_items` is using snake case while all other properties are using camel case.

### CODE REVIEW

**Comment by thomasyu888** [comment]
@vpchung Thanks for this - we actually want to move more towards snakecase in the python client as it is more of a Python standard.  Happy to discuss more here though.

**Comment by vpchung** [comment]
Oh, nice -  I'm all for that transition!

**Comment by thomasyu888** [comment]
@vpchung Does that mean we close this PR?

File: synapseclient/table.py [hunk 1]
```diff
@@ -866,13 +866,13 @@ class Dataset(SchemaBase):
...
     _synapse_entity_type: str = "org.sagebionetworks.repo.model.table.Dataset"
-    _property_keys: List[str] = SchemaBase._property_keys + ['dataset_items']
+    _property_keys: List[str] = SchemaBase._property_keys + ['datasetItems']
     _local_keys: List[str] = SchemaBase._local_keys + ['folders_to_add', 'force']
 
     def __init__(self, name=None, columns=None, parent=None, properties=None,
                  annotations=None, local_state=None, dataset_items=None,
-                 folders=None, force=None, **kwargs):
-        self.properties.setdefault('dataset_items', [])
+                 folder_ids=None, force=None, **kwargs):
```
Comment (thomasyu888) [inline synapseclient/table.py:874]:
```suggestion
                 folder=None, force=None, **kwargs):
```

File: synapseclient/table.py [hunk 2]
```diff
@@ -866,13 +866,13 @@ class Dataset(SchemaBase):
...
     _synapse_entity_type: str = "org.sagebionetworks.repo.model.table.Dataset"
-    _property_keys: List[str] = SchemaBase._property_keys + ['dataset_items']
+    _property_keys: List[str] = SchemaBase._property_keys + ['datasetItems']
     _local_keys: List[str] = SchemaBase._local_keys + ['folders_to_add', 'force']
 
     def __init__(self, name=None, columns=None, parent=None, properties=None,
                  annotations=None, local_state=None, dataset_items=None,
-                 folders=None, force=None, **kwargs):
-        self.properties.setdefault('dataset_items', [])
+                 folder=None, force=None, **kwargs):
```
Comment (thomasyu888) [inline synapseclient/table.py:874]:
```suggestion
                 folders=None, force=None, **kwargs):
```

**Pull request merged**